public/js/custom-analytics.js

```diff
@@ -0,0 +1,110 @@
+jQuery(function($) {
+
+  // Debug flag
+  var debugMode = true;
+
+  // Default time delay before checking location
+  var callBackTime = 100;
+
+  // # px before tracking a reader
+  var readerLocation = 150;
+
+  // Set some flags for tracking & execution
+  var timer = 0;
+  var scroller = false;
+  var endOpeningTimes = false;
+  var endPatientsSay = false;
+  var endSurgeryGPs = false;
+  var endServices = false;
+  var endParking = false;
+  var didComplete = false;
+
+  // Set some time variables to calculate reading time
+  var startTime = new Date();
+  var beginning = startTime.getTime();
+  var totalTime = 0;
+
+  // Get some information about the current page
+  var pageTitle = document.title;
+
+  function scrolledSection(start, section) {
+    currentTime = new Date();
+    if (start) {
+      scrollStart = currentTime.getTime();
+      timeToScroll = Math.round((scrollStart - beginning) / 1000);
+      if (!debugMode) {
+          ga('send', 'event', section, timeToScroll+'s');
+      } else {
+          alert(section + ' (Time: ' + timeToScroll + 's)');
+      }
+
+    } else {
+      scrollEnd = currentTime.getTime();
+      timeToScroll = Math.round((scrollEnd - scrollStart) / 1000);
+      if (!debugMode) {
+          ga('send', 'event', 'User reached the "' + section +'" section', timeToScroll+'s');
+      } else {
+          alert('User reached the "' + section + '" section (Time: ' + timeToScroll + 's)');
+      }
+    }
+    scroller = true;
+  }
+
+  // Check the location and track user
+  function trackLocation() {
+      bottom = $(window).height() + $(window).scrollTop();
+      height = $(document).height();
+      results = $('.reading-width .sr-only').attr('data-results');
+      results_14 = Math.round(results / 4);
+      results_12 = Math.round(results / 2);
+      results_34 = Math.round(results_14 * 3);
+
+
+      // If user starts to scroll send an event
+      if (bottom > readerLocation && !scroller) {
+        scrolledSection(true, 'User started scrolling');
+        scroller = true;
+      }
+
+      if ($('h2.opening-times:contains("GP")').length && bottom > $('h2.opening-times:contains("GP")').offset().top + $('h2.opening-times:contains("GP")').outerHeight() && !endOpeningTimes) {
```

Comment:
You could also move the logic in all but the last 'if' into a function, something like:

```
  function hasScrolledToSection(bottom, selector, ended) {
      var element = $(selector);
     return element.length && bottom > element.offset().top + element.outerHeight() && !ended;
  }
```

which you'd call like:
`if (hasScrolledToSection(bottom, 'h2.opening-times:contains("GP")', endOpeningTimes)) {`

I've put the $ selector in a local variable in the sample function as otherwise jquery will scan the page each time you call the $(), so this will improve performance slightly